xwiki-platform-core/xwiki-platform-rendering/xwiki-platform-rendering-macros/xwiki-platform-rendering-macro-include/src/main/java/org/xwiki/rendering/internal/macro/include/IncludeMacro.java

```diff
@@ -202,7 +202,23 @@ public void setDocumentDisplayer(DocumentDisplayer documentDisplayer)
                 references.pop();
             }
         }
-
+        
+        // Exclude First Heading
+        if (parameters.excludeFirstHeading()) {
+            if (result.getChildren().size() > 0) {
+                Block firstBlock = result.getChildren().get(0);
+                if (firstBlock instanceof SectionBlock && firstBlock.getChildren().size() > 0) {
+                    Block sectionFirstBlock = firstBlock.getChildren().get(0);
+                    if (sectionFirstBlock instanceof HeaderBlock) {
+                        List<Block> contentWithoutheading = firstBlock.getChildren().subList(1,
```

Comment:
why sublist? (never used this api btw). Why not getting the children of  the HeaderBlock?

Reply:
HeaderBlock.getChildren() will give the content of Header only but we want to extract the content of the whole section. sublist will extract the content without the HeaderBlock.

Reply:
Right. I meant HeaderBlock.getNextSibling() :)

Reply:
Sounds good. I'll apply the PR and fix the "if"s if need be, a bit later tonight. Thx